### ISSUE TITLE
Redo secret key APIs

### ISSUE THREAD

**Issue by benthecarman** [p2pderivatives/rust-dlc#185]
Currently there are two APIs around getting secret keys `get_new_secret_key` and `get_secret_key_for_pubkey`. These make it hard for wallets to manage because the only real way to implement this is to have an index of public keys to private keys and store them. It would be much better if `get_new_secret_key` gave the user an identifier and then they could use this for generating the key statelessly so storage isn't needed for every key generated.

**Comment by Tibo-lg** [comment]
I feel the best might be to just get rid of the signing within the library and have something like `create_adaptor_signature` on the `Signer` interface. Then there'd be no need for these two methods anymore I think, and it would give more flexibility for users. What do you think?

Backward compatibility is not yet something that should be considered to much for now I think.

**Comment by benthecarman** [comment]
Yeah that makes sense, you still need to give pubkey before the signature however so there needs to be some interface for that.

**Comment by Tibo-lg** [comment]
For sure, I guess having a signer per contract with a `get_funding_pubkey` or something like that should work. It becomes a bit more tricky for channels, you'd probably need to extend the interface (like having a `DlcChannelSigner` that extends the base `DlcSigner`).
